numpy/core/records.py

```diff
@@ -662,6 +662,8 @@ def fromrecords(recList, dtype=None, shape=None, formats=None, names=None,
 
     if formats is None and dtype is None:  # slower
         obj = sb.array(recList, dtype=object)
+        if sum(obj.shape) == 0:
```

Comment:
Can you maybe just go with the safe `obj.shape == (0,)`, and leave the other cases to be dealt with in future?

Reply:
done in 430f598